src/ClientHandle.cpp

```diff
@@ -1831,7 +1831,33 @@ void cClientHandle::HandleUnmount(void)
 void cClientHandle::HandleTabCompletion(const AString & a_Text)
 {
 	AStringVector Results;
-	m_Player->GetWorld()->TabCompleteUserName(a_Text, Results);
+	if (cRoot::Get()->GetServer()->ShouldAllowMultiWorldTabCompletion())
+	{
+		class cWorldCallback : public cWorldListCallback
+		{
+		public:
+			cWorldCallback(AStringVector & a_Results, const AString & a_Search) :
+				m_Results(a_Results),
+				m_Search(a_Search)
+			{
+			}
+
+			virtual bool Item(cWorld * a_World) override
+			{
+				a_World->TabCompleteUserName(m_Search, m_Results);
+				return true;
+			}
+		private:
+			AStringVector & m_Results;
+			const AString & m_Search;
+		} WC(Results, a_Text);
+
+		cRoot::Get()->ForEachWorld(WC);
```

Comment:
Extract this into a separate function in `cRoot`, so that we can export it to Lua API later on when we open up the tab completion to plugins completely.